Sources/Core/Presenter/UploadManager.swift

```diff
@@ -121,5 +141,7 @@ public final class UploadManager {
                 }
             })
             .share()
+        
+        return item.uploading ?? .empty()
```

Comment:
What would cause `item.uploading` being `nil` here?

Reply:
That's a good question. `UploadingItem` is creating without uploading a subscription. Then `UploadManager` called for `startUploading(item:)` it will set a shared observable for the item as `.uploading`. Another reason, that a model could be created with an attachment and no need to upload it.

Reply:
I referred to this very line (L145). You assign a the `uploading` value on L117 and then on L145 it can be `nil` again and you use ` ?? .empty()` as a fallback value. I think it will never be `nil`. That's why I asked about why it can be `nil`, becase the code suggests it's valid for it to be `nil`.

Reply:
@VojtaStavik  I think this is a question of coding style. In this specific case, I need to set a value to an optional property and at the same time return it as a non-optional value. So, I was lazy, I know that I can return `.empty()`. In other cases probably I won't have it and I need to make a local var and return it. So, what do you think?

Reply:
We don't need to complicate it and use just `return item.uploading!`. We're 100% sure the value is there (we create it on the previous line). If `item.uploading` was `nil` here, it would mean our assumptions are terribly wrong and we should know about it anyway :)

Reply:
>  If item.uploading was nil here, it would mean our assumptions are terribly wrong and we should know about it anyway :)
The problem is that our customers will know about that first. The usual case in our tests and Example app everything is working as expected, but our customers could start complaining that on production with millions of people something wrong. It would be really bad if it will be a crash.

Reply:
We're both talking hypothetical situations so it doesn't matter. Leave it as is.